src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -351,7 +351,7 @@ public static Expression<TDelegate> Lambda<TDelegate>(Expression body, String na
         public static Expression<TDelegate> Lambda<TDelegate>(Expression body, String name, bool tailCall, IEnumerable<ParameterExpression> parameters)
         {
             var parameterList = parameters.ToReadOnly();
-            ValidateLambdaArgs(typeof(TDelegate), ref body, parameterList);
+            ValidateLambdaArgs(typeof(TDelegate), ref body, parameterList, nameof(TDelegate));
```

Comment:
I think ArgumentException is appropriate: take a look at the non-generic overloads. For these, we are testing a parameter (named delegateType). Also, I think its common to validate the generic parameter and throw an ArgumentException (e.g. Activator.CreateInstance)